src/fsharp/NameResolution.fs

```diff
@@ -2014,16 +2016,17 @@ let DecodeFSharpEvent (pinfos:PropInfo list) ad g (ncenv:NameResolver) m =
 
 /// Returns all record label names for the given type.
 let GetRecordLabelsForType g nenv typ =
-    if isRecdTy g typ then
-        let typeName = NicePrint.minimalStringOfType nenv.eDisplayEnv typ
-        nenv.eFieldLabels
-        |> Seq.filter (fun kv -> 
-            kv.Value 
-            |> List.exists (fun r -> r.TyconRef.DisplayName = typeName))
-        |> Seq.map (fun kv -> kv.Key)
-        |> Set.ofSeq
-    else
-        Set.empty
+    (if isRecdTy g typ then
```

Comment:
Instead of () + pipe. Maybe introduce a temp variable

Reply:
Yeah, I already didn't like it as I was typing it :) I just rewrote it with a CE.